### PR TITLE
Add github native code scanning

### PR DESCRIPTION
**Pull request by alazymeme**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable
- [x] Documentation in docs/ or install-docs/ was updated, if applicable

This implements githubs new code scanning stuff. By default, the workflow scans:

- Every push to the default branch
- Every pull request to the default branch
- The default branch at 3pm every Sunday

More information can be found below:
https://docs.github.com/en/free-pro-team@latest/github/finding-security-vulnerabilities-and-errors-in-your-code/about-code-scanning
https://docs.github.com/en/free-pro-team@latest/github/finding-security-vulnerabilities-and-errors-in-your-code/configuring-code-scanning

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    # ℹ️ Command-line programs to run using the OS shell.
+    # 📚 https://git.io/JvXDl
+
+    # ✏️ If the Autobuild fails above, remove it and uncomment the following three lines
+    #    and modify them (or add more) to build your code if your project
+    #    uses a compiled language
+
+    #- run: |
+    #   make bootstrap
+    #   make release
```
Comment (RAnders00) [inline .github/workflows/codeql-analysis.yml:68]: Remove autobuild since we dont need it

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+name: "CodeQL"
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [master]
+  schedule:
+    - cron: '0 2 * * 5'
```
Comment (RAnders00) [inline .github/workflows/codeql-analysis.yml:15]: Add a comment that this means "3pm every Sunday"

**Review by RAnders00 — CHANGES_REQUESTED** [review]

**Review by RAnders00 — APPROVED** [review]

**Comment by RAnders00** [comment]
Is the CodeQL analyze going to run on every PR (and take so long)?

**Review by pajlada — APPROVED** [review]
:+1:

**Comment by alazymeme** [comment]
> Is the CodeQL analyze going to run on every PR (and take so long)?

Yes - I can alternatively configure it so that it only runs on every push/weekly so it doesn't hold up the merging of prs?

**Comment by RAnders00** [comment]
It's fine since it's not set to be "required" to merge a pull request. As far as I understand it doesn't fail the status check either when the PR introduces a vulnerability, right?

**Pull request merged**